wgengine/magicsock/magicsock.go

```diff
@@ -3291,6 +3313,46 @@ func (c *Conn) UpdateStatus(sb *ipnstate.StatusBuilder) {
 	})
 }
 
+// updateStats updates the statistics counters with the src, dst, and cnts.
+// It is a noop if statistics gathering is not enabled.
+func (c *Conn) updateStats(src netip.Addr, dst netip.AddrPort, cnts tunstats.Counts) {
+	// This method is written to be inlineable.
+	if c.stats.enabled.Load() {
+		c.updateStatsEnabled(src, dst, cnts)
+	}
+}
+
+func (c *Conn) updateStatsEnabled(src netip.Addr, dst netip.AddrPort, cnts tunstats.Counts) {
+	conn := flowtrack.Tuple{Src: netip.AddrPortFrom(src, 0), Dst: dst}
+	c.stats.mu.Lock()
+	defer c.stats.mu.Unlock()
+	if c.stats.m == nil {
+		c.stats.m = make(map[flowtrack.Tuple]tunstats.Counts)
+	}
+	c.stats.m[conn] = c.stats.m[conn].Add(cnts)
```

Comment:
or `mak.Set(&c.stats.m, conn, c.stats.m[conn].Add(cnts))` instead of these 4 lines

Reply:
Hahaha... should this be added to the `maps` package?

Reply:
Feel free to propose it! ;)